.github/workflows/ci.yml

```diff
@@ -0,0 +1,147 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+      - release-*
+  pull_request: {}
+  workflow_dispatch:
+    inputs:
+      version:
+        description: Package version (e.g. v0.1.0)
+        required: false
```

Comment:
I'm not sure everyone will want to take on the hassle of maintaining release branches, but they'll presumably at least want to tag and push versioned releases of their Function. So the goal here is:

* When the CI workflow runs against a PR or the main branch we build and push it as `v0.0.0-gitsha`.
* You can run the CI workflow manually and provide whatever version you want.

I don't love that the `v0.0.0-gitsha` pattern loses semver context, but I don't really want to inflict assumptions around how git tags are used on folks.

Reply:
Yeah i'm a bit conflicted on this one too - I think the intent to not require release branches is reasonable, but I wonder if we can push folks towards using tags at least, so we can get a meaningful version number always.

The build submodule uses tags if the exist and if they don't then it falls back to `v0.0.0`:
https://github.com/upbound/build/blob/master/makelib/common.mk#L214-L225

Is that type of conditional logic possible here?